### PR TITLE
Search articles functionallity (task #2160)

### PR DESCRIPTION
**Pull request by flangofas**
ss-cms-articles-search

### CODE REVIEW

File: src/Model/Table/ArticlesTable.php [hunk 1]
```diff
@@ -211,6 +211,53 @@ public function findRelated(Query $query, array $options)
...
 
     /**
+     * Search function which search the given fields.
+     * Options:
+     * - term, the search term - required
+     * - fields, It is the schema fields could be a string or array - required
+     *
+     * @param  Query  $query   [description]
+     * @param  array  $options [description]
+     * @return [type]          [description]
```
Comment (georgeconstantinou) [inline src/Model/Table/ArticlesTable.php]: @flangofas missing description

File: src/Controller/ArticlesController.php [hunk 2]
```diff
@@ -17,14 +17,29 @@ class ArticlesController extends AppController
...
+        if (is_null($search)) {
+            if ($articles->isEmpty()) {
+                $this->Flash->set(__('No articles were found. Please add one.'));
+                return $this->redirect(['action' => 'add']);
+            }
+        } else {
+            $articles = $articles->find(
+                'search',
+                [
+                    'fieldNames' => ['title', 'slug'],
```
Comment (mamchenkov) [inline src/Controller/ArticlesController.php]: @flangofas excerpt and body content as well please

Comment (flangofas) [inline src/Controller/ArticlesController.php]: @mamchenkov they are not show on the index page. is it OK?

**Comment by flangofas** [comment]
@georgeconstantinou please recheck.

**Pull request merged**